### ISSUE TITLE
RLS/Clippy doesn't lint newly created files

### ISSUE THREAD

**Issue by stinodego** [rust-lang/vscode-rust#921]
Whenever I create a new .rs file in my project, I don't receive any warnings/lints until I restart VSCode. After I restart, everything works as expected.

I'm sure I must be doing something wrong; could someone help me out?

**Comment by lnicola** [comment]
Can you try https://marketplace.visualstudio.com/items?itemName=matklad.rust-analyzer instead (though it might be a bit finicky to set up in WSL).

**Comment by stinodego** [comment]
Sure, I'll give that a shot! Will report back.

**Comment by stinodego** [comment]
> Can you try https://marketplace.visualstudio.com/items?itemName=matklad.rust-analyzer instead (though it might be a bit finicky to set up in WSL -- you need to install it on the remote side and make sure everything is in `PATH`).

Yes, if I disable the Rust extension and enable rust-analyzer instead, everything works as expected. Installing was easy by the way, no need to mess around with `PATH` or anything.

How does this extension compare to the "Rust" extension otherwise?

**Comment by lnicola** [comment]
See https://rust-analyzer.github.io/blog/2020/04/20/first-release.html and https://rust-lang.github.io/compiler-team/working-groups/rls-2.0/. It's supposed to replace RLS at some point, but it's currently developed outside of the Rust project.

> How does this extension compare to the "Rust" extension otherwise?

Favourably, in my (biased) opinion :-).

**Comment by stinodego** [comment]
I'll still keep this issue open for now, as it seems relevant to the Rust extension.

My first impression of `rust-analyzer` is that it tries to do too much; I prefer simplicity. But that's just my very first impression, I might give it a shot later.

**Comment by lnicola** [comment]
You can disable most features, if that's what you mean, check out the manual or file an issue on that repo.
